tests/perf/benchmark.py

```diff
@@ -0,0 +1,200 @@
+        eval_upto (str): The last serial operation to evaluate. Choose on of ('train', 'export', 'optimize').
+            Operations include the preceeding ones.
+            e.x) Eval up to 'optimize': train -> eval -> export -> eval -> optimize -> eval
+            Default to 'train'.
+        output_root (str): Output root dirctory for logs and results. Defaults to './otx-benchmark'.
+        dry_run (bool): Whether to just print the OTX command without execution. Defaults to False.
+        tags (dict): Key-values pair metadata for the experiment. Defaults to {}.
+    """
+
+    def __init__(
+        self,
+        datasets: List[str],
+        data_root: str = "data",
+        num_epoch: int = 0,
+        num_repeat: int = 1,
+        train_params: dict = {},
+        track_resources: bool = False,
+        eval_upto: str = "train",
+        output_root: str = "otx-benchmark",
+        dry_run: bool = False,
+        tags: dict = {},
+        **kwargs,
+    ):
+        self.datasets = datasets
+        self.data_root = data_root
+        self.num_epoch = num_epoch
+        self.num_repeat = num_repeat
+        self.train_params = train_params
+        self.track_resources = track_resources
+        self.eval_upto = eval_upto
+        self.output_root = output_root
+        self.dry_run = dry_run
+        self.tags = tags
+        self.subset_dir_names = kwargs.get("subset_dir_names", {"train": "", "val": "", "test": ""})
+
+    def run(
+        self,
+        model_id: str,
+        train_params: dict = {},
+        tags: dict = {},
+    ) -> pd.DataFrame:
+        """Run configured benchmark with given model and return the result.
+
+        Args:
+            model_id (str): Target model identifier
+            train_params (dict): Overrides global benchmark train params
+            tags (dict): Overrides global benchmark tags
+
+        Retruns:
+            pd.DataFrame: Table with benchmark metrics
+        """
+
+        # Build config file
+        cfg = self._build_config(model_id, train_params, tags)
+        cfg_dir = Path(cfg["output_path"])
+        cfg_dir.mkdir(parents=True, exist_ok=True)
+        cfg_path = cfg_dir / "cfg.yaml"
+        with open(cfg_path, "w") as cfg_file:
+            yaml.dump(cfg, cfg_file, indent=2)
+        cmd = [
+            "python",
+            "tools/experiment.py",
+            "-f",
+            cfg_path,
+        ]
+        if self.dry_run:
+            cmd.append("-d")
+        # Run benchmark
+        check_run(cmd)
+        # Load result
+        result = self.load_result(cfg_dir)
+        return result
+
+    @staticmethod
+    def load_result(result_path: str) -> pd.DataFrame | None:
+        """Load benchmark results recursively and merge as pd.DataFrame.
+
+        Args:
+            result_path (str): Result directory or speicific file.
+
+        Retruns:
+            pd.DataFrame: Table with benchmark metrics & options
```

Comment:
```suggestion
            pd.DataFrame | None : Table with benchmark metrics & options
```
How about adding None in comments?

Reply:
Fixed in https://github.com/openvinotoolkit/training_extensions/pull/2742/commits/91bfec13541e70258511473e29b8b56dfed4c3d1